homwork.rb

```diff
@@ -0,0 +1,219 @@
+students = [
+  {
+      first_name: 'Ahmed',
+      last_name: 'Al-Bahrani'
+  },
+  {
+      first_name: 'Reem',
+      last_name: 'AlHarbi',
+  },
+  {
+      first_name: 'Mansour',
+      last_name: 'Almohsen',
+  }
+]
+
+upper_case_full_names = []
+students.each do |name|
+
+    upper_case_full_names.insert(
+
+        "#{name[:first_name].upcase} #{name[:last_name].upcase}")
+end
+
+////////////////////////////////////////////////////////////////////////////////////////
+
+users = [
+  {
+      name: 'Hanan',
+      orders: [
+          {
+              description: 'a bike'
+          }
+      ]
+  },
+  {
+      name: 'Abdulrahman',
+      orders: [
+          {
+              description: 'bees'
+          },
+          {
+              description: 'fishing rod'
+          }
+      ]
+  },
+  {
+      name: 'Turki',
+      orders: [
+          {
+              description: 'a MacBook'
+          },
+          {
+              description: 'The West Wing DVDs'
+          },
+          {
+              description: 'headphones'
+          },
+          {
+              description: 'a kitten'
+          }
+      ]
+  }
+]
+
+first_order_for_each_user = []
+
+users.each do |nameorders| 
+
+    first_order_for_each_user.insert(name[:orders]. description)
+
+    end
```

Comment:
- You have `nameorders` in your block, but then you're using `name`, so your program doesn't recognize where name is coming from.

- `name[:orders]` refers to the array of orders for each user, the question requires the first order only.

- `. description` doesn't work with ruby, it will think `.description` is a method